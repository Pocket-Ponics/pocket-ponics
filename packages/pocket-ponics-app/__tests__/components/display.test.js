import React from 'react'
import { shallow } from 'enzyme'
import { TouchableOpacity } from 'react-native'

import Display from '../../components/greenhouse/display'

describe('Greenhouse Display', () => {
	test('renders correctly', () => {
		const tree = shallow(
			<Display 
				navigation={{}} 
				tiers={[
					{
						'tier': 1,
						'plant_id': 5,
						'ph_level': 0,
						'ec_level': 0,
						'water_level': 0,
						'cycle_time': null,
						'num_plants': 0
					},
					{
						'tier': 2,
						'plant_id': 2,
						'ph_level': 0,
						'ec_level': 0,
						'water_level': 0,
						'cycle_time': null,
						'num_plants': 0
					},
					{
						'tier': 3,
						'plant_id': 3,
						'ph_level': 0,
						'ec_level': 0,
						'water_level': 0,
						'cycle_time': null,
						'num_plants': 0
					},
					{
						'tier': 4,
						'plant_id': 4,
						'ph_level': 0,
						'ec_level': 0,
						'water_level': 0,
						'cycle_time': null,
						'num_plants': 0
					}
				]}/>).html()
		expect(tree).toMatchSnapshot()
	})

	test('renders with no tiers', () => {
		const tree = shallow(
			<Display 
				navigation={{}} 
				tiers={[
					{
						'tier': 1,
						'plant_id': null,
					},
					{
						'tier': 2,
						'plant_id': null,
					},
					{
						'tier': 3,
						'plant_id': null,
					},
					{
						'tier': 4,
						'plant_id': null,
					}
				]}/>).html()
		expect(tree).toMatchSnapshot()
	})

	test('renders with seedlings', () => {
		const tree = shallow(
			<Display 
				navigation={{}} 
				displaySeedlings={true}
				tiers={[
					{
						'tier': 1,
						'plant_id': null,
					},
					{
						'tier': 2,
						'plant_id': null,
					},
					{
						'tier': 3,
						'plant_id': null,
					},
					{
						'tier': 4,
						'plant_id': null,
					}
				]}/>).html()
		expect(tree).toMatchSnapshot()
	})

	test('renders correctly', () => {
		const navigation = {
			navigate: jest.fn()
		}

		const tree = shallow(
			<Display 
				navigation={navigation} 
				displaySeedlings={true}
				tiers={[
					{
						'tier': 1,
						'plant_id': 5,
						'ph_level': 0,
						'ec_level': 0,
						'water_level': 0,
						'cycle_time': null,
						'num_plants': 0
					},
					{
						'tier': 2,
						'plant_id': 2,
						'ph_level': 0,
						'ec_level': 0,
						'water_level': 0,
						'cycle_time': null,
						'num_plants': 0
					},
					{
						'tier': 3,
						'plant_id': 3,
						'ph_level': 0,
						'ec_level': 0,
						'water_level': 0,
						'cycle_time': null,
						'num_plants': 0
					},
					{
						'tier': 4,
						'plant_id': 4,
						'ph_level': 0,
						'ec_level': 0,
						'water_level': 0,
						'cycle_time': null,
						'num_plants': 0
					}
				]}/>)

		tree.find(TouchableOpacity).at(0).props().onPress()
		tree.find(TouchableOpacity).at(1).props().onPress()
		tree.find(TouchableOpacity).at(2).props().onPress()
		tree.find(TouchableOpacity).at(3).props().onPress()
		tree.find(TouchableOpacity).at(4).props().onPress()

		expect(navigation.navigate).toHaveBeenCalledTimes(5)
	})
})