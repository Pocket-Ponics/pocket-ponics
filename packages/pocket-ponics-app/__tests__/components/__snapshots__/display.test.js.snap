// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Greenhouse Display renders correctly 1`] = `"<View style=\\"color:#472600;align-items:center;justify-content:center;margin:10px;flex:1\\"><View style=\\"flex:1.8;width:100%;align-items:center;justify-content:center;opacity:1\\"><View style=\\"flex:1;flex-direction:row;justify-content:center;aspect-ratio:1.1904761904761905px\\"><Image source=\\"[object Object]\\" style=\\"0:[object Object];1:[object Object];2:[object Object]\\"></Image><Image source=\\"[object Object]\\" style=\\"height:65%;width:100%;margin-top:10%;resize-mode:contain\\"></Image></View></View><View style=\\"flex:1;width:100%;align-items:center;justify-content:center;opacity:1\\"><View style=\\"flex:1;flex-direction:row;justify-content:center;aspect-ratio:2.142857142857143px\\"><Image source=\\"[object Object]\\" style=\\"0:[object Object];1:[object Object];2:[object Object]\\"></Image><Image source=\\"[object Object]\\" style=\\"flex:1;aspect-ratio:1.5px;resize-mode:contain;margin-top:6%\\"></Image><Image source=\\"[object Object]\\" style=\\"flex:1;aspect-ratio:1.5px;resize-mode:contain;margin-top:6%\\"></Image><Image source=\\"[object Object]\\" style=\\"flex:1;aspect-ratio:1.5px;resize-mode:contain;margin-top:6%\\"></Image></View></View><View style=\\"flex:1;width:100%;align-items:center;justify-content:center;opacity:1\\"><View style=\\"flex:1;flex-direction:row;justify-content:center;aspect-ratio:2.142857142857143px\\"><Image source=\\"[object Object]\\" style=\\"0:[object Object];1:[object Object];2:[object Object]\\"></Image><Image source=\\"[object Object]\\" style=\\"flex:1;aspect-ratio:1.5px;resize-mode:contain;margin-top:6%\\"></Image><Image source=\\"[object Object]\\" style=\\"flex:1;aspect-ratio:1.5px;resize-mode:contain;margin-top:6%\\"></Image><Image source=\\"[object Object]\\" style=\\"flex:1;aspect-ratio:1.5px;resize-mode:contain;margin-top:6%\\"></Image></View></View><View style=\\"flex:1;width:100%;align-items:center;justify-content:center;opacity:1\\"><View style=\\"flex:1;flex-direction:row;justify-content:center;aspect-ratio:2.142857142857143px\\"><Image source=\\"[object Object]\\" style=\\"0:[object Object];1:[object Object];2:[object Object]\\"></Image><Image source=\\"[object Object]\\" style=\\"flex:1;aspect-ratio:1.5px;resize-mode:contain;margin-top:6%\\"></Image><Image source=\\"[object Object]\\" style=\\"flex:1;aspect-ratio:1.5px;resize-mode:contain;margin-top:6%\\"></Image><Image source=\\"[object Object]\\" style=\\"flex:1;aspect-ratio:1.5px;resize-mode:contain;margin-top:6%\\"></Image></View></View></View>"`;

exports[`Greenhouse Display renders with no tiers 1`] = `"<View style=\\"color:#472600;align-items:center;justify-content:center;margin:10px;flex:1\\"><View style=\\"flex:1.8;width:100%;align-items:center;justify-content:center;opacity:1\\"><View style=\\"flex:1;flex-direction:row;justify-content:center;aspect-ratio:1.1904761904761905px\\"><Image source=\\"[object Object]\\" style=\\"0:[object Object];1:[object Object];2:[object Object]\\"></Image></View></View><View style=\\"flex:1;width:100%;align-items:center;justify-content:center;opacity:1\\"><View style=\\"flex:1;flex-direction:row;justify-content:center;aspect-ratio:2.142857142857143px\\"><Image source=\\"[object Object]\\" style=\\"0:[object Object];1:[object Object];2:[object Object]\\"></Image></View></View><View style=\\"flex:1;width:100%;align-items:center;justify-content:center;opacity:1\\"><View style=\\"flex:1;flex-direction:row;justify-content:center;aspect-ratio:2.142857142857143px\\"><Image source=\\"[object Object]\\" style=\\"0:[object Object];1:[object Object];2:[object Object]\\"></Image></View></View><View style=\\"flex:1;width:100%;align-items:center;justify-content:center;opacity:1\\"><View style=\\"flex:1;flex-direction:row;justify-content:center;aspect-ratio:2.142857142857143px\\"><Image source=\\"[object Object]\\" style=\\"0:[object Object];1:[object Object];2:[object Object]\\"></Image></View></View></View>"`;

exports[`Greenhouse Display renders with seedlings 1`] = `"<View style=\\"color:#472600;align-items:center;justify-content:center;margin:10px;flex:1\\"><View style=\\"flex:1.8;width:100%;align-items:center;justify-content:center;opacity:1\\"><View style=\\"flex:1;flex-direction:row;justify-content:center;aspect-ratio:1.1904761904761905px\\"><Image source=\\"[object Object]\\" style=\\"0:[object Object];1:[object Object];2:[object Object]\\"></Image></View></View><View style=\\"flex:1;width:100%;align-items:center;justify-content:center;opacity:1\\"><View style=\\"flex:1;flex-direction:row;justify-content:center;aspect-ratio:2.142857142857143px\\"><Image source=\\"[object Object]\\" style=\\"0:[object Object];1:[object Object];2:[object Object]\\"></Image></View></View><View style=\\"flex:1;width:100%;align-items:center;justify-content:center;opacity:1\\"><View style=\\"flex:1;flex-direction:row;justify-content:center;aspect-ratio:2.142857142857143px\\"><Image source=\\"[object Object]\\" style=\\"0:[object Object];1:[object Object];2:[object Object]\\"></Image></View></View><View style=\\"flex:1;width:100%;align-items:center;justify-content:center;opacity:1\\"><View style=\\"flex:1;flex-direction:row;justify-content:center;aspect-ratio:2.142857142857143px\\"><Image source=\\"[object Object]\\" style=\\"0:[object Object];1:[object Object];2:[object Object]\\"></Image></View></View><View style=\\"flex:1;width:100%;align-items:center;justify-content:center;opacity:1\\"><View style=\\"flex:1;flex-direction:row;justify-content:center;aspect-ratio:2.142857142857143px\\"><Image source=\\"[object Object]\\" style=\\"0:[object Object];1:[object Object];2:[object Object]\\"></Image><Image source=\\"[object Object]\\" style=\\"flex:1;aspect-ratio:1px;resize-mode:contain;margin-top:8%\\"></Image><Image source=\\"[object Object]\\" style=\\"flex:1;aspect-ratio:1px;resize-mode:contain;margin-top:8%\\"></Image><Image source=\\"[object Object]\\" style=\\"flex:1;aspect-ratio:1px;resize-mode:contain;margin-top:8%\\"></Image><Image source=\\"[object Object]\\" style=\\"flex:1;aspect-ratio:1px;resize-mode:contain;margin-top:8%\\"></Image><Image source=\\"[object Object]\\" style=\\"flex:1;aspect-ratio:1px;resize-mode:contain;margin-top:8%\\"></Image></View></View></View>"`;
